scripts/hal_alignment/test_hal_gab_access.pl

```diff
@@ -0,0 +1,258 @@
+
+=item B<[--hal_dir hal_directory]>
+
+The path to the directory containing the HAL files. Can also be specified through the B<COMPARA_HAL_DIR>
+environmental variable.
+
+=item B<[--out output_json]>
+
+Name of the output JSON file.
+
+=back
+
+=cut
+
+use Getopt::Long;
+use Text::CSV qw(csv);
+use Proc::ProcessTable;
+use JSON;
+
+use Bio::EnsEMBL::Registry;
+use Bio::EnsEMBL::Compara::DBSQL::DBAdaptor;
+
+my $help;
+my $reg_conf;
+my $compara = 'compara_curr';
+my $mlss_id;
+my $regions_tsv;
+my $outfile = "test_hal_gab_access.json";
+my $hal_dir;
+
+GetOptions(
+    'help'          => \$help,
+    'reg_conf=s'    => \$reg_conf,
+    'compara=s'     => \$compara,
+    'mlss=i'        => \$mlss_id,
+    'regions=s'     => \$regions_tsv,
+    'hal_dir=s'     => \$hal_dir,
+    'out=s'         => \$outfile,
+);
+
+# Print Help and exit if help is requested
+if ($help) {
+    use Pod::Usage;
+    pod2usage({-exitvalue => 0, -verbose => 2});
+}
+
+# Process command line parameters:
+die("The MLSS ID must be specified!") if !$mlss_id;
+die("The regions TSV file must be specified!") if !$regions_tsv;
+
+if ($hal_dir && $ENV{COMPARA_HAL_DIR}) {
+    die("Cannot set HAL dir as the COMPARA_HAL_DIR envirnomental variable is already set!");
+} elsif ($hal_dir) {
+    $ENV{COMPARA_HAL_DIR} = $hal_dir;
+}
+
+$reg_conf = $ENV{COMPARA_REG_PATH} if !$reg_conf;
+
+# Load all test regions from TSV file:
+sub load_regions {
+    my $infile = shift;
+    my $aoh = csv (in => $infile, headers => "auto", sep_char => "\t");
+    for my $r (@$aoh) {
+        $r->{Region} = "$r->{Species}:$r->{CoordSys}:$r->{DnaFrag}:$r->{Start}:$r->{End}";
+    }
+    return $aoh;
+}
+
+#################################################
+## Get the adaptors from the Registry
+
+our $registry = 'Bio::EnsEMBL::Registry';
+$registry->load_all($reg_conf, 0, 0, 0, 'throw_if_missing') if $reg_conf;
+
+my $compara_dba;
+if ($compara =~ /mysql:\/\//) {
+    $compara_dba = Bio::EnsEMBL::Compara::DBSQL::DBAdaptor->new(-url=>$compara);
+} else {
+    $compara_dba = Bio::EnsEMBL::Registry->get_DBAdaptor($compara, 'compara');
+}
+if (!$compara_dba) {
+  die "Cannot connect to compara database <$compara>.";
+}
+
+our $dnafrag_adaptor = $compara_dba->get_DnaFragAdaptor();
+our $gab_adaptor = $compara_dba->get_GenomicAlignBlockAdaptor();
+our $gdb_adaptor = $compara_dba->get_GenomeDBAdaptor();
+our $mlss_adaptor = $compara_dba->get_MethodLinkSpeciesSetAdaptor();
+
+# Get MLSS obejct by ID:
+our $cactus_mlss = $mlss_adaptor->fetch_by_dbID($mlss_id);
+
+# Load test regions:
+my $test_regions =  load_regions($regions_tsv);
+
+my $res = {};
+
+# Test access via a slice:
+sub test_slice_access {
+    my $reg = shift;
+    my $res = shift;
+
+    my $sliceAdaptor = $registry->get_adaptor($reg->{Species}, 'Core', 'Slice');
+    my $slice = $sliceAdaptor->fetch_by_region($reg->{CoordSys}, $reg->{DnaFrag}, $reg->{Start}, $reg->{End});
```

Comment:
This is two suggestions in one:
1. to use snake case in the name of `$slice_adaptor`, for consistency with the rest of the script.
2. to set the default coordinate system to `'toplevel'`, with the idea of making it easier to create our own test region TSV file independently of `sample_genomic_regions.pl` if the need arises.

Please let me know if you like one suggestion but not the other; I'm happy to remove whatever you're not happy with.
```suggestion
    my $slice_adaptor = $registry->get_adaptor($reg->{Species}, 'Core', 'Slice');
    my $coord_sys = $reg->{CoordSys} || 'toplevel';
    my $slice = $slice_adaptor->fetch_by_region($coord_sys, $reg->{DnaFrag}, $reg->{Start}, $reg->{End});
```